apiserver/facades/client/application/deploy.go

```diff
@@ -263,7 +271,18 @@ func (api *APIBase) addUnits(
 		if err != nil {
 			return nil, internalerrors.Errorf("parsing unit name %q: %w", unit.Name(), err)
 		}
-		if err := api.applicationService.AddUnits(ctx, domainapplication.StorageParentDir, appName, applicationservice.AddUnitArg{UnitName: unitName}); err != nil {
+
+		var unitPlacement *instance.Placement
+		if i < len(placement) {
+			unitPlacement = placement[i]
+		}
+
+		unitArg := applicationservice.AddUnitArg{
+			UnitName:  unitName,
+			Placement: unitPlacement,
+		}
+
+		if err := api.applicationService.AddUnits(ctx, domainapplication.StorageParentDir, appName, unitArg); err != nil {
```

Comment:
Why aren't any tests breaking? was this only tested in `base_test.go`?

Reply:
Because the default placement was always the same. So it was a no-op.